src/WebApi.OutputCache.V2/CacheOutputAttribute.cs

```diff
@@ -175,6 +181,8 @@ public override void OnActionExecuting(HttpActionContext actionContext)
 
             if (!_webApiCache.Contains(cachekey)) return;
 
+            if (IsNoCacheHeaderInRequest(actionContext)) return;
```

Comment:
can you move this up? there is no need to go through cache key generator if this is the case.
I'd put it right after `IsCachingAllowed()`